spec/hyrax/transactions/create_work_spec.rb

```diff
@@ -83,6 +83,40 @@
     end
   end
 
+  context 'when requesting a lease' do
+    let(:after)     { 'restricted' }
+    let(:during)    { 'open' }
+    let(:end_date)  { (Time.zone.today + 2).to_s }
+    let(:request)   { Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_LEASE }
+    let(:step_args) { { apply_visibility: [visibility: request, release_date: end_date, during: during, after: after] } }
+
+    it 'sets the lease' do
+      expect { transaction.with_step_args(step_args).call(work) }
+        .to change { work.lease }
+        .to be_a_lease_matching(release_date: end_date, during: during, after: after)
+    end
+  end
+
+  context 'when requesting an embargo' do
+    let(:after)     { 'open' }
+    let(:during)    { 'restricted' }
+    let(:end_date)  { (Time.zone.today + 2).to_s }
+    let(:request)   { Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_EMBARGO }
+    let(:step_args) { { apply_visibility: [visibility: request, release_date: end_date, during: during, after: after] } }
+
+    it 'sets the embargo' do
+      expect { transaction.with_step_args(step_args).call(work) }
+        .to change { work.embargo }
+        .to be_an_embargo_matching(release_date: end_date, during: during, after: after)
+    end
+  end
+
+  context 'when requesting an embargo' do
```

Comment:
This context looks like a duplicate block.  Should it be about setting visibility (apart from lease or embargo) or can it just be removed?

Reply:
I've removed this block, thanks!